### ISSUE TITLE
Increase robustness when scanning for content

### ISSUE THREAD

**Issue by Abestanis** [nt4f04uNd/sweyer#127]
### Suggestion

To prevent issues like #122, we should:
- [ ] Be more robust and don't fail the entire content search because one item fails. We should catch exceptions on an item level and ignore them.
- [ ] Catch errors in the content search and show an error UI instead of hanging forever.
- [ ] Handle more fields being null. It looks like all columns in the Android media database can based on the [create statement here](https://cs.android.com/android/platform/superproject/main/+/main:packages/providers/MediaProvider/src/com/android/providers/media/DatabaseHelper.java;l=1108-1144) and the fact that the album database is just [a view into the `files` database](https://cs.android.com/android/platform/superproject/main/+/main:packages/providers/MediaProvider/src/com/android/providers/media/DatabaseHelper.java;l=1606-1622), so we should be more resilient against nulls.

**Comment by Abestanis** [comment]
Also, off topic, shouldn't we undraft https://github.com/nt4f04uNd/sweyer/releases/tag/untagged-88d5c1dd963c02bce466 ?

**Comment by nt4f04uNd** [comment]
> Also, off topic, shouldn't we undraft https://github.com/nt4f04uNd/sweyer/releases/tag/untagged-88d5c1dd963c02bce466 ?

True

**Comment by nt4f04uNd** [comment]
Your suggestions seem reasonable

**Comment by nt4f04uNd** [comment]
> It looks like all columns in the Android media database can be null

This is crazy to me, by the way

**Comment by Abestanis** [comment]
Yes, we already improved the situation immensely by only ignoring individual content items when they are missing any data that we require, but ideally we would handle all missing non-essential data.

For example, currently a song without a title will be ignored, but we should accept it and just show a "<Unnamed song>" or something like that. I'll go through and create sub-issues for all the attributes where we should accept `null` values.
